korean-air/src/components/flightInfoPage/FlightCard.tsx

```diff
@@ -0,0 +1,302 @@
+                <CommonPrice>78,300원</CommonPrice>
+                <CommonSeat>7석</CommonSeat>
+              </BtnLayout>
+
+              <NotSelectBtnLayout>
+                <NotSelectedTitle>프레스티지</NotSelectedTitle>
+                <NotSelectedSub>미운영</NotSelectedSub>
+              </NotSelectBtnLayout>
+            </BtnGroup>
+          </div>
+          <div>
+            <ClickSection className={`${!click ? "click" : ""}`}>
+              <div>
+                <section>
+                  <IcChangeFlight />
+                  <CommonSeat>변경 불가</CommonSeat>
+                </section>
+                <section>
+                  <IcRefundFlight />
+                  <CommonSeat>환불 7,000원</CommonSeat>
+                </section>
+              </div>
+              <div>
+                <section>
+                  <IcShortInfoFlight />
+                  <CommonSeat>
+                    좌석 승급 불가/ (결제 페이지에서 확인 가능)
+                  </CommonSeat>
+                </section>
+              </div>
+            </ClickSection>
+          </div>
+        </article>
+      </Wrapper>
+    </>
+  );
+};
+
+export default FlightCard;
+
+const Wrapper = styled.section`
+  display: flex;
+  flex-direction: column;
+  width: 100%;
+  height: 29.1rem;
+  margin-top: 2rem;
+  border-radius: 10px;
+  background: ${theme.colors.white};
+  box-shadow: 0rem 0rem 2rem 0rem rgba(107, 117, 136, 0.2);
+  border-radius: 1rem;
+
+  & > article {
+    display: flex;
+    flex-direction: column;
+    width: 100%;
+    flex-shrink: 0;
+    justify-content: space-between;
+    /* background-color: pink; */
+    border-radius: 1rem;
+  }
+
+  & > article > section {
+    margin-left: 2rem;
+  }
+
+  & > article > div {
+    display: flex;
+    justify-content: space-around;
+    align-items: center;
+    width: 100%;
+    height: 7.4rem;
+    flex-shrink: 0;
+  }
+`;
+
+const FlightTime = styled.p`
+  color: ${theme.colors.black};
+  ${theme.fonts.title_bold_18};
+`;
+
+const Area = styled.p`
+  color: ${theme.colors.grey_2};
+  ${theme.fonts.body_medium_12};
+`;
+
+const ExpectedTimeLayout = styled.article`
+  display: inline-block;
+  flex-direction: column;
+  align-items: center;
+  margin-top: 1rem;
+`;
+
+const Time = styled.p`
+  color: ${theme.colors.navy};
+  ${theme.fonts.body_medium_12};
+  margin-left: 2rem;
+`;
+
+const JinairInfo = styled.p`
+  color: ${theme.colors.grey_2};
+  ${theme.fonts.body_medium_12};
+`;
+
+const JinairLayout = styled.section`
+  width: 100%;
+  height: 2.5rem;
+  background-color: green;
+`;
+
+const BtnLayout = styled.section`
+  display: flex;
+  width: 8.3rem;
+  height: 8.3rem;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  gap: 0.4rem;
+  flex-shrink: 0;
+  cursor: pointer;
+  border-radius: 10px;
+  border: 0.1rem solid ${theme.colors.grey_4};
+  background: ${theme.colors.white};
+
+  &.click {
+    background: ${theme.colors.navy};
+  }
```

Comment:
```suggestion
  background: ${ click ? theme.colors.navy : theme.colors.white };
```

그럼 이제 이 부분은 이렇게 바꿔줄 수 있겠지?! 
삼항 연산자가 익숙하지 않을 수 있으니까 설명하자면 저 코드는 click 값이 true면 navy 색으로, false면 white 색으로 지정해줘~ 라는 뜻이야. 

그럼 여기서 끝이 아니고, 
우리는 JS가 아니라 TS를 사용해주고 있기 때문에 아마 이렇게만 적용시켜주면 위에서 props를 내려주는 곳에서 빨간 줄이 뜰거야! 그건 넘겨주는 props에 대한 타입 지정을 해주지 않았기 때문인데, **styled-components에게 넘겨주는 props는 타입지정을 반드시 해줘야하고**, 해주는 방식은 다음과 같아.
```js
const BtnLayout = styled.section<{click : boolean}>`
  // ...
` 
```
이렇게 어떤 element 써줄지 지정해주는 것과 백틱 사이에 대괄호를 넣어주고 그안에 타입을 지정해주는 중괄호를 넣어주면 돼! 

관련된 자료 링크 달아놓을테니까 꼭꼭 읽어보고 잘 써보자 ! 
[블로그 링크 : TS에서 styled components props 쓰기](https://thsd-stjd.tistory.com/134)

Reply:
나도 참고 링크 열심히 읽어볼게 ! 짱짱